tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/GrypeTask.java

```diff
@@ -2,31 +2,74 @@
 
 import org.gradle.api.DefaultTask;
 import org.gradle.api.tasks.TaskAction;
-import org.gradle.internal.impldep.com.esotericsoftware.kryo.io.ByteBufferOutputStream;
+import org.gradle.api.tasks.options.Option;
 import org.gradle.process.ExecOperations;
-import org.gradle.process.ExecResult;
 
 import javax.inject.Inject;
 import java.io.*;
 import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
 import java.util.*;
-import java.util.function.Predicate;
 
-import static org.graalvm.internal.tck.DockerUtils.getAllowedImages;
+import static org.graalvm.internal.tck.DockerUtils.extractImagesNames;
+import static org.graalvm.internal.tck.DockerUtils.getAllAllowedImages;
 
 public abstract class GrypeTask extends DefaultTask {
 
     @Inject
     protected abstract ExecOperations getExecOperations();
 
+    @Option(option = "baseCommit", description = "Last commit from master")
+    void setBaseCommit(String baseCommit) {
+        this.baseCommit = baseCommit;
+    }
+
+    @Option(option = "newCommit", description = "HEAD commit of the pull request")
+    void setNewCommit(String newCommit) {
+        this.newCommit = newCommit;
+    }
+
+    private String newCommit;
+    private String baseCommit;
+
     private final String jqMatcher = " | jq -c '.matches | .[] | .vulnerability | select(.severity | (contains(\"High\") or contains(\"Critical\")))'";
 
+    private List<URL> getChangedImages(String base, String head){
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        getExecOperations().exec(spec -> {
+            spec.setStandardOutput(baos);
+            spec.commandLine("git", "diff", "--name-only", "--diff-filter=ACMRT", base, head);
```

Comment:
Hang on, isn't this only to check whether a file in `tests/tck-build-logic/src/main/resources/allowed-docker-images` has changed? If so, I'd be for duplicating the workflow: one for PRs and one for weekly checks. Or why do we need to maintain this code here?

Reply:
No, this is actually to get all files that are changed or added. With workflows we only know that "something" changed on specified path, but we don't know what files have changed. This way we will get a list of changed/added files from allowed-docker-images and execute check only for them. We don't need to execute for example `nats` docker image scan if we only bumped `mysql` image version. On the other hand, we will execute scan of all docker images once a week to check if new vulnerabilities are discovered (but that won't block PRs that change only one image).